src/Request.ts

```diff
@@ -178,3 +178,46 @@ export function isRetryable (status?: number): boolean {
       ].indexOf(status) !== -1)
     )
 }
+
+export function fetch(endpoint: string): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    const parsedUrl = url.parse(endpoint)
+
+    const req = (parsedUrl.protocol === 'https:' ? https : http).get(endpoint, res => {
+      res.pipe(concat((bodyBuffer: Buffer) => {
+        if (res.statusCode === 200) {
+          return resolve(bodyBuffer.toString())
+        }
+
+        const err = new UploadError(`HTTP status ${res.statusCode} received from bundle server`)
+        err.responseText = bodyBuffer.toString()
+
+        if (!isRetryable(res.statusCode)) {
+          err.isRetryable = false
+        }
+
+        if (res.statusCode && (res.statusCode >= 400 && res.statusCode < 500)) {
+          err.code = UploadErrorCode.MISC_BAD_REQUEST
+        } else {
+          err.code = UploadErrorCode.SERVER_ERROR
+        }
+
+        return reject(err)
+      }))
+    })
+
+    req.on('error', e => {
+      const err = new UploadError('Unknown connection error')
+      err.cause = e
+      err.code = UploadErrorCode.UNKNOWN
+      reject(err)
+    })
+
+    req.setTimeout(TIMEOUT_MS, () => {
```

Comment:
This could probably be a lot shorter than `TIMEOUT_MS` because it relates to a period of socket inactivity, rather than the length of time the request is taking.

Worth introducing a new variable `IDLE_TIMEOUT_MS` and rolling it out to `send()` too?

Reply:
Isn't this already an idle timeout? So I think we rename `TIMEOUT_MS` to `IDLE_TIMEOUT_MS` and add a new `TIMEOUT_MS` for the entire request length

Reply:
Actually, if it's a timeout for the entire request then it'd need to be fairly long — maybe we just want to reduce the current timeout's default?